misk/src/main/kotlin/misk/web/resources/StaticResourceAction.kt

```diff
@@ -0,0 +1,142 @@
+package misk.web.resources
+
+import misk.resources.ResourceLoader
+import misk.scope.ActionScoped
+import misk.security.authz.Unauthenticated
+import misk.web.Get
+import misk.web.Post
+import misk.web.Request
+import misk.web.RequestContentType
+import misk.web.Response
+import misk.web.ResponseBody
+import misk.web.ResponseContentType
+import misk.web.actions.NotFoundAction
+import misk.web.actions.WebAction
+import misk.web.mediatype.MediaTypes
+import misk.web.toResponseBody
+import okhttp3.Headers
+import okhttp3.HttpUrl
+import okhttp3.MediaType
+import okio.BufferedSink
+import okio.BufferedSource
+import java.net.HttpURLConnection
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * StaticResourceAction
+ *
+ * This data class is used with Guice multibindings. Register instances by calling `multibind()`
+ * in a `KAbstractModule`:
+ *
+ * ```
+ * multibind<StaticResourceEntry>().toInstance(StaticResourceEntry(...))
+ * ```
+ */
+
+@Singleton
+class StaticResourceAction : WebAction {
+  @Inject private lateinit var entries: List<StaticResourceEntry>
+  @Inject @JvmSuppressWildcards private lateinit var clientRequest: ActionScoped<Request>
+  @Inject private lateinit var resourceLoader: ResourceLoader
+
+  @Get("/{path:.*}")
+  @Post("/{path:.*}")
+  @RequestContentType(MediaTypes.ALL)
+  @ResponseContentType(MediaTypes.ALL)
+  @Unauthenticated  // TODO(adrw) should this be unauthenticated?
+  fun action(): Response<ResponseBody> {
+    val request = clientRequest.get()
+    return getResponse(request)
+  }
+
+  private fun getResponse(request: Request): Response<ResponseBody> {
+    val urlPath = request.url.encodedPath()
+    val matchedEntry =
+        ResourceEntryCommon.findEntryFromUrl(entries, request.url) as StaticResourceEntry?
+    return when (exists(normalizePath(urlPath))) {
+      Kind.NO_MATCH -> when {
+        !urlPath.contains(".") && !urlPath.endsWith("/") ->
```

Comment:
what’s with the `.` ?

Reply:
I don’t think we need to treat filename-like things as special. Although it’s handy to not treat `/foo.txt` as the same as `/foo`, it might be a pain if we have a directory like `/dr.who`